From 4033907045abcf498686c1b34be7b0160260088d Mon Sep 17 00:00:00 2001
From: Andreas Henriksson <andreas@fatal.se>
Date: Sat, 02 Oct 2010 12:51:54 +0000
Subject: sftp: fix poll() timeout.

When switching from select() to poll() in commit
"sftp: Use poll() to cope with openssh-5.6 changes" (c6be45c8934)
the difference in seconds vs milliseconds for select/poll timeout
argument was missed.

SFTP_READ_TIMEOUT is defined in seconds, so multiply it with 1000
when using it with poll().

https://bugzilla.gnome.org/show_bug.cgi?id=631169
---
diff --git a/daemon/gvfsbackendsftp.c b/daemon/gvfsbackendsftp.c
index 9fecf6a..f6b7785 100644
--- a/daemon/gvfsbackendsftp.c
+++ b/daemon/gvfsbackendsftp.c
@@ -869,7 +869,7 @@ handle_login (GVfsBackend *backend,
       fds[1].fd = prompt_fd;
       fds[1].events = POLLIN;
       
-      ret = poll(fds, 2, SFTP_READ_TIMEOUT);
+      ret = poll(fds, 2, SFTP_READ_TIMEOUT * 1000);
       
       if (ret <= 0)
         {
--
cgit v0.8.3.1
